File: config/paths.py
"""Portable paths for the PDAC P1 analysis repository."""

from __future__ import annotations

import os
from pathlib import Path


REPOSITORY_ROOT = Path(__file__).resolve().parents[1]
PROJECT_ROOT = Path(os.environ.get("PDAC_PROJECT_ROOT", REPOSITORY_ROOT)).expanduser().resolve()
RAW_DATA_ROOT = PROJECT_ROOT / "data"
EXTERNAL_DATA_ROOT = PROJECT_ROOT / "data_external"
ANALYSIS_ROOT = PROJECT_ROOT / "analysis"
RESULTS_ROOT = PROJECT_ROOT / "results"
FIGURE_ROOT = ANALYSIS_ROOT / "15_priority_figures"


def describe_paths() -> dict[str, Path]:
    return {
        "repository_root": REPOSITORY_ROOT,
        "project_root": PROJECT_ROOT,
        "raw_data_root": RAW_DATA_ROOT,
        "external_data_root": EXTERNAL_DATA_ROOT,
        "analysis_root": ANALYSIS_ROOT,
        "results_root": RESULTS_ROOT,
        "figure_root": FIGURE_ROOT,
    }

File: config/paths.R
pdac_paths <- function() {
  repo_root <- Sys.getenv("PDAC_CODE_REPO_ROOT", unset = getwd())
  project_root <- Sys.getenv("PDAC_PROJECT_ROOT", unset = repo_root)
  repo_root <- normalizePath(repo_root, winslash = "/", mustWork = FALSE)
  project_root <- normalizePath(project_root, winslash = "/", mustWork = FALSE)
  list(
    repo_root = repo_root,
    project_root = project_root,
    raw_data_root = file.path(project_root, "data"),
    external_data_root = file.path(project_root, "data_external"),
    analysis_root = file.path(project_root, "analysis"),
    results_root = file.path(project_root, "results"),
    figure_root = file.path(project_root, "analysis", "15_priority_figures")
  )
}

pdac_script_repo_root <- function() {
  args <- commandArgs(trailingOnly = FALSE)
  file_arg <- grep("^--file=", args, value = TRUE)
  if (length(file_arg)) {
    script_path <- normalizePath(sub("^--file=", "", file_arg[[1]]), winslash = "/", mustWork = FALSE)
    return(normalizePath(file.path(dirname(script_path), "../.."), winslash = "/", mustWork = FALSE))
  }
  normalizePath(getwd(), winslash = "/", mustWork = FALSE)
}
